management-service/services/experiment_service.go

```diff
@@ -555,6 +551,40 @@ func (svc *experimentService) save(exp *models.Experiment) (*models.Experiment,
 	return svc.GetDBRecord(exp.ProjectID, exp.ID)
 }
 
+func (svc *experimentService) filterFieldValues(query *gorm.DB, params ListExperimentsParams) (*gorm.DB, error) {
+	if params.Fields != nil && len(*params.Fields) != 0 {
+		err := validateListExperimentFieldNames(*params.Fields)
+		if err != nil {
+			return nil, err
+		}
+
+		// Stores a map of unique field names which are unique column names to be selected in the db query
+		fieldNamesSet := make(map[string]struct{})
+		for _, field := range *params.Fields {
+			fieldName := field
+			// Add ExperimentFieldStatus to the query because status_friendly does not exist in the db as a column
+			if field == models.ExperimentFieldStatusFriendly {
+				// Add ExperimentFieldStartTime and ExperimentFieldEndTime to the query as they are required for
+				// determining the statusFriendly field; these two fields may or may not be returned depending on
+				// what is actually specified in params.Fields
+				fieldNamesSet[string(models.ExperimentFieldStartTime)] = struct{}{}
+				fieldNamesSet[string(models.ExperimentFieldEndTime)] = struct{}{}
+
+				fieldName = models.ExperimentFieldStatus
+			}
+			fieldNamesSet[string(fieldName)] = struct{}{}
+		}
+
+		// Retrieve a slice of unique strings from fieldNamesSet; query.Select only accepts []string
+		var fieldNames []string
+		for k := range fieldNamesSet {
+			fieldNames = append(fieldNames, k)
+		}
+		query = query.Select(fieldNames)
```

Comment:
This `fieldFieldValues` function essentially specifies `SELECT` values in the DB query such that only columns corresponding to the field values requested get retrieved.